backend/remote/backend_common.go

```diff
@@ -415,25 +415,27 @@ func (b *Remote) checkPolicy(stopCtx, cancelCtx context.Context, op *backend.Ope
 			return fmt.Errorf(msgPrefix + " hard failed.")
 		case tfe.PolicySoftFailed:
 			if op.Type == backend.OperationTypePlan || op.UIOut == nil || op.UIIn == nil ||
-				op.AutoApprove || !pc.Actions.IsOverridable || !pc.Permissions.CanOverride {
+				!pc.Actions.IsOverridable || !pc.Permissions.CanOverride {
 				return fmt.Errorf(msgPrefix + " soft failed.")
 			}
 		default:
 			return fmt.Errorf("Unknown or unexpected policy state: %s", pc.Status)
 		}
 
-		opts := &terraform.InputOpts{
-			Id:          "override",
-			Query:       "\nDo you want to override the soft failed policy check?",
-			Description: "Only 'override' will be accepted to override.",
-		}
+		if !op.AutoApprove {
+			opts := &terraform.InputOpts{
+				Id:          "override",
+				Query:       "\nDo you want to override the soft failed policy check?",
+				Description: "Only 'override' will be accepted to override.",
+			}
 
-		err = b.confirm(stopCtx, op, opts, r, "override")
-		if err != nil && err != errRunOverridden {
-			return err
+			err = b.confirm(stopCtx, op, opts, r, "override")
+			if err != nil && err != errRunOverridden {
+				return err
+			}
 		}
```

Comment:
Looks good; can we move this logic up in to the `PolicySoftFailed` `case` block with the unoveriddable logic though? I think the flow felt a little weird even before these changes, with the logic around soft failures being split like this.

Reply:
Agreed, moving it up